### PR TITLE
No docs removed modules

### PR DESCRIPTION
**Pull request by abadger**
##### SUMMARY
Once modules are removed, remove the docs for them as well.

Change the removed module stubs  to point people towards the old docs in their error message instead.

##### ISSUE TYPE
 - Bugfix Pull Request
 - Docs Pull Request

##### COMPONENT NAME
Multiple removed modules

##### ANSIBLE VERSION

```
2.7.0
```

### CODE REVIEW

File: test/sanity/validate-modules/main.py [hunk 1]
```diff
@@ -1001,8 +887,167 @@ def _validate_docs(self):
...
+                # metadata has that information.
+                if 'removed' in metadata['status']:
+                    removed = True
+                if 'deprecated' in metadata['status']:
+                    deprecated = True
+                if (deprecated or removed) and len(metadata['status']) > 1:
+                    self.reporter.error(
+                        path=self.object_path,
+                        code=333,
+                        msg='ANSIBLE_METADATA with status "deprecated" or "removed" cannot have any other status'
```
Comment (gundalow) [inline test/sanity/validate-modules/main.py]: Suggestion to make it clearer that `status` is an attribute
`ANSIBLE_METADATA.status must be exactly one of "deprecated" or "removed"'`

File: test/sanity/validate-modules/main.py [hunk 2]
```diff
@@ -831,135 +831,21 @@ def _validate_docs_schema(self, doc, schema, name, error_code):
...
-                    # This module has an alias, which we can tell as it's a symlink
-                    # Rather than checking for `module: $filename` we need to check against the true filename
-                    self._validate_docs_schema(doc, doc_schema(os.readlink(self.object_path).split('.')[0]), 'DOCUMENTATION', 305)
-                else:
-                    # This is the normal case
-                    self._validate_docs_schema(doc, doc_schema(self.object_name.split('.')[0]), 'DOCUMENTATION', 305)
-
-                self._check_version_added(doc)
-                self._check_for_new_args(doc)
+        documentation_exists = False
```
Comment (gundalow) [inline test/sanity/validate-modules/main.py:835]: +1 to these variables, makes it a lot easier to read, thanks.

File: test/sanity/validate-modules/main.py [hunk 3]
```diff
@@ -1001,8 +887,167 @@ def _validate_docs(self):
...
+                # time.  We can figure that out for deprecated but we can't for removed.  Only the
+                # metadata has that information.
+                if 'removed' in metadata['status']:
+                    removed = True
+                if 'deprecated' in metadata['status']:
+                    deprecated = True
+                if (deprecated or removed) and len(metadata['status']) > 1:
+                    self.reporter.error(
+                        path=self.object_path,
+                        code=333,
```
Comment (gundalow) [inline test/sanity/validate-modules/main.py:902]: Please add to `docs/docsite/rst/dev_guide/testing_validate-modules.rst`

**Review by gundalow — COMMENTED** [review]
Minor suggestions only. Looks good, thanks for doing this

**Review by gundalow — APPROVED** [review]

File: docs/docsite/rst/porting_guides/porting_guide_2.3.rst [hunk 4]
```diff
@@ -118,15 +118,23 @@ Deprecation notices
...
+* cl_interface_policy
+* cl_license
+* cl_ports
+* nxos_mtu use :ref:`nxos_system <nxos_system_module>` instead
+
+.. note::
+
+    These modules may no longer have documentation in the current release.  Please see the
+    `Ansible 2.3 module documentation
+    <https://docs.ansible.com/ansible/2.3list_of_all_modules.html>`_ if you need
```
Comment (acozine) [inline docs/docsite/rst/porting_guides/porting_guide_2.3.rst]: missing a / here - `2.3/list_of . . .`

Comment (abadger) [reply docs/docsite/rst/porting_guides/porting_guide_2.3.rst]: Fixed, thanks.

**Comment by abadger** [comment]
merged to the devel branch for the 2.7.0 release.

**Pull request merged**